### ISSUE TITLE
Nearest Neighbour Upsampling for 3D images

### ISSUE THREAD

**Issue by shubhamjain0594** [soumith/cudnn.torch#269]
Is there a layer implemented for Nearest Neighbour Upsampling for 3D Images?

**Comment by ngimel** [comment]
No, but you can repurpose average pooling for that. Pooling backward would correspond to upsampling forward, and vice versa. PRs welcome.
